packages/ui/src/components/Table/components/RemoteUpdateTable/index.tsx

```diff
@@ -0,0 +1,212 @@
+/** *******************************************************************************************************************
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+  
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  
+      http://www.apache.org/licenses/LICENSE-2.0
+  
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.                                                                              *
+ ******************************************************************************************************************** */
+import { FC, useMemo, useState, useCallback, useEffect } from 'react';
+import TableComponent, { TableProps as CloudscapeTableProps } from '@cloudscape-design/components/table';
+import Pagination, { PaginationProps } from '@cloudscape-design/components/pagination';
+import TextFilter, { TextFilterProps } from '@cloudscape-design/components/text-filter';
+import Header from '@cloudscape-design/components/header';
+import Button from '@cloudscape-design/components/button';
+import { NonCancelableEventHandler } from '@cloudscape-design/components/internal/events';
+import { useDebouncedCallback } from 'use-debounce';
+
+import EmptyState from '../EmptyState';
+import { RemoteUpdateTableProps, InternalTableProps, FetchDataOptions } from '../../types';
+import {
+    DEFAULT_TRACK_BY,
+    DEFAULT_LOADING_TEXT,
+    DEFAULT_PAGINATION_LABELS,
+    DEFAULT_FILTERING_PLACEHOLDER,
+    DEFAULT_FILTERING_ARIA_LABEL,
+    getAriaLabels,
+    DEFAULT_PAGE_SIZE,
+} from '../../config';
+
+const DEFAULT_DEBOUNCE_TIMER = 500;
+
+const RemoteUpdateTable: FC<RemoteUpdateTableProps & InternalTableProps> = ({
```

Comment:
Nice this looks really useful :)

Reply:
Thanks. We are also planning to add a DynamoDB Specific table to allow users to do previous page/next page without pagination for DynamoDB scan with next token.